### ISSUE TITLE
iMX8M Mini EVK intermittently hangs during boot

### ISSUE THREAD

**Issue by diegorusso** [ARMmbed/meta-mbl#590]
### Description

It has been discovered that sometimes iMX8M Mini EVK hangs during boot with the following error:

```
U-Boot SPL 2018.03-imx_v2018.03_4.14.78_1.0.0_ga-mbl+g4cac377 (Jun 17 2019 - 17:04:03 +0000)
power_bd71837_init
DDRINFO: start lpddr4 ddr init
DRAM PHY training for 3000MTS
check ddr4_pmu_train_imem code
check ddr4_pmu_train_imem code pass
check ddr4_pmu_train_dmem code
check ddr4_pmu_train_dmem code pass
Training PASS
DRAM PHY training for 3000MTS
check ddr4_pmu_train_imem code
check ddr4_pmu_train_imem code pass
check ddr4_pmu_train_dmem code
check ddr4_pmu_train_dmem code pass
Training PASS
DRAM PHY training for 400MTS
check ddr4_pmu_train_imem code
check ddr4_pmu_train_imem code pass
check ddr4_pmu_train_dmem code
check ddr4_pmu_train_dmem code pass
Training PASS
DRAM PHY training for 100MTS
check ddr4_pmu_train_imem code
check ddr4_pmu_train_imem code pass
check ddr4_pmu_train_dmem code
check ddr4_pmu_train_dmem code pass
Training PASS
DDRINFO:ddrphy calibration done
DDRINFO: ddrmix config done
Normal Boot
Trying to boot from MMC1
NOTICE: BL2: v1.5(release):p9.0.0_1.0.2-auto-beta-20181219-17-gba3f2f0
NOTICE: BL2: Built : 17:18:37, Jun 17 2019
INFO: Using crypto library 'mbed TLS'
INFO: BL2: Doing platform setup
INFO: BL2: Loading image id 3
INFO: Loading image id=7 at address 0x900000
INFO: Image id=7 loaded: 0x900000 - 0x90060e
INFO: Loading image id=9 at address 0x900000
INFO: Image id=9 loaded: 0x900000 - 0x9004da
INFO: Loading image id=13 at address 0x900000
INFO: Image id=13 loaded: 0x900000 - 0x9003e9
INFO: Loading image id=3 at address 0x900000
INFO: Image id=3 loaded: 0x900000 - 0x90d169
INFO: BL2: Loading image id 4
INFO: Loading image id=10 at address 0xbe000000
INFO: Image id=10 loaded: 0xbe000000 - 0xbe0004e8
INFO: Loading image id=14 at address 0xbe000000
INFO: Image id=14 loaded: 0xbe000000 - 0xbe000487
INFO: Loading image id=4 at address 0xbe000000
INFO: Image id=4 loaded: 0xbe000000 - 0xbe00001c
INFO: OPTEE ep=0xbe000000
INFO: OPTEE header info:
INFO: magic=0x4554504f
INFO: version=0x2
INFO: arch=0x1
INFO: flags=0x0
INFO: nb_images=0x1
INFO: BL2: Loading image id 21
INFO: Loading image id=21 at address 0xbe000000
INFO: Image id=21 loaded: 0xbe000000 - 0xbe050410
INFO: BL2: Skip loading image id 22
INFO: BL2: Loading image id 5
INFO: Loading image id=11 at address 0x40200000
INFO: Image id=11 loaded: 0x40200000 - 0x402004ea
INFO: Loading image id=15 at address 0x40200000
INFO: Image id=15 loaded: 0x40200000 - 0x402003f9
INFO: Loading image id=5 at address 0x40200000
INFO: Image id=5 loaded: 0x40200000 - 0x402a2a75
NOTICE: BL2: Booting BL31
INFO: Entry point address = 0x900000
INFO: SPSR = 0x3cd
NOTICE: Configuring TZASC380
INFO: enter
INFO: security_inversion_en 0
INFO: exit
NOTICE: RDC off 
NOTICE: BL31: v1.5(release):p9.0.0_1.0.2-auto-beta-20181219-17-gba3f2f0
NOTICE: BL31: Built : 17:18:25, Jun 17 2019
INFO: GICv3 with legacy support detected. ARM GICV3 driver initialized in EL3
INFO: BL31: Initializing runtime services
NOTICE: sip svc init
INFO: BL31: Initializing BL32
E/TC:0 0 Panic 'Failed to add OP-TEE Device Tree node' at core/arch/arm/kernel/generic_boot.c:952 <update_external_dt>
E/TC:0 0 Call stack:
E/TC:0 0 0x00000000be00c2f8
```
This happens intermittently and it is not systematic.
This affects both mbl-os-0.6 and mbl-os-0.7.

### Issue request type

    [ ] Question
    [ ] Enhancement
    [x] Bug

**Comment by ciarmcom** [comment]
*Internal Jira reference: https://jira.arm.com/browse/MBOCUSTRIA-1300*

**Comment by jjohnson-arm** [comment]
Fixed by commit c51deda4ed35ea8dc31795c27af15fd42fa687ff on warrior-dev. Will be part of mbl-os-0.8 release.
